framework-plugins/lisk-framework-forger-plugin/src/webhooks.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import axios, { AxiosError } from 'axios';
+import * as Debug from 'debug';
+
+import { Webhook } from './types';
+// eslint-disable-next-line new-cap
+const debug = Debug('plugin:forger:webhooks');
+
+interface HttpHeaders {
+	[key: string]: string;
+}
+
+interface BlockCreated {
+	readonly height: number;
+	readonly reward: string;
+	readonly forgerAddress: string;
+}
+
+interface BlockMissed {
+	readonly height: number;
+	readonly missedBlocksByAddress: { [key: string]: number };
+}
+
+interface NodeStatusChange {
+	readonly reason: string;
+}
+
+interface WebHookPayload {
+	readonly event: string;
+	readonly time: number;
+	readonly payload: BlockCreated | BlockMissed | NodeStatusChange;
+}
+
+export class Webhooks {
+	private readonly headers: HttpHeaders;
+	private readonly registeredEvents: readonly Webhook[];
+
+	public constructor(defaultHeaders: HttpHeaders, configuredEvents: readonly Webhook[]) {
+		this.headers = defaultHeaders;
+		this.registeredEvents = configuredEvents;
+	}
+
+	public async execute(eventData: WebHookPayload, targetURL: string): Promise<object> {
+		return axios.post(targetURL, eventData, { headers: this.headers });
+	}
+
+	public async handleEvent(data: WebHookPayload): Promise<void> {
+		const requiredEvents = [];
+		for (const aRegisteredEvent of this.registeredEvents) {
+			if (aRegisteredEvent.events.includes(data.event)) {
+				requiredEvents.push({ url: aRegisteredEvent.url, data });
+			}
+		}
+
+		try {
+			for (const anEvent of requiredEvents) {
+				// eslint-disable-next-line no-void
+				await this.execute(anEvent.data, anEvent.url);
+			}
+		} catch (err) {
+			debug('Error during webhook processing');
```

Comment:
prefer to have 1 debug line

Reply:
The problem is that Axios adds special properties to the err object outside of message so if the error is caused by a successful http interaction with the receiving end a single line will not show it. Or maybe you meant to show both variables using one call?

Reply:
if it's one debug, call then showing it is fine

Reply:
Done